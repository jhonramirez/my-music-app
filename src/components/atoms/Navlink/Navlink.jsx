import { NavLink } from "react-router-dom";

export const Navlink = ({ navigate, name, img, alt, className }) => {
  return (
    <>
      {name ? (
        <NavLink className={className} to={`${navigate}`}>
          {name}
        </NavLink>
      ) : (
        ""
      )}
      {img ? (
        <NavLink to={`${navigate}`}>
          <img className={className} alt={alt} src={img} />
        </NavLink>
      ) : (
        ""
      )}
    </>
  );
};
