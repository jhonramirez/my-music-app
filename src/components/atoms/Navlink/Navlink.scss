.navbar {
  display: flex;
  flex-direction: column;
  padding: 0.5rem;
  color: #dad4ce;
  text-decoration: none;
  align-items: flex-start;
}

.navbar:hover {
  color: #ffffff; 
}

.img__home {
  width: 25px;
  height: 25px;
}

.img__login {
  width: 300px;
  height: 300px;
}
