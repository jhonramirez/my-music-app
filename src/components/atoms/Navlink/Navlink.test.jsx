import { render, screen } from "@testing-library/react";
import { Navlink } from "./Navlink";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";

describe("Pruebas en <Navlink />", () => {
  const { container } = render(
    <MemoryRouter>
      <Navlink name={"proof"} className="className" />
      <Navlink img={"img"} className="className" />
    </MemoryRouter>
  );
  const NavlinkElement = screen.getByText("proof");
  const testImage = document.querySelector("img");

  test("Render Navlink y match con el snapshot", () => {
    expect(NavlinkElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("Se evalua que tenga un className", () => {
    expect(NavlinkElement).toHaveClass("className");
  });

  test("Cuando se envia una imagen se evalua que tenga un src", () => {
    expect(testImage.src).toContain("img");
  });
});
