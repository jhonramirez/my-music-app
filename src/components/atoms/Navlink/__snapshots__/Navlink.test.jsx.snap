// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navlink /> Render Navlink y match con el snapshot 1`] = `
<div>
  <a
    class="className"
    href="/undefined"
  >
    proof
  </a>
  <a
    class=""
    href="/undefined"
  >
    <img
      class="className"
      src="img"
    />
  </a>
</div>
`;
